src/shadowfax.py

```diff
@@ -56,17 +56,34 @@ async def shadowfax(message, guid, logger):
     to find nodes that occur in publications with our input nodes, then finding
     paths that connect our input nodes through these intermediate nodes."""
     qgraph = message["message"]["query_graph"]
-    pinned_node_ids = []
+    pinned_node_ids_set = set()
     for node in qgraph["nodes"].values():
         if node.get("ids", None) is not None:
-            pinned_node_ids.append(node["ids"][0])
-        else:
-            unpinned_node_category = node.get("categories", ["biolink:NamedThing"])[0]
-    if len(pinned_node_ids) != 2:
-        logger.info(f"{guid}: Pathfinder queries require two pinned nodes.")
+            pinned_node_ids_set.add(node["ids"][0])
+    if len(pinned_node_ids_set) != 2:
+        logger.error(f"{guid}: Pathfinder queries require two different pinned nodes.")
         return message, 500
+    pinned_node_ids = list(pinned_node_ids_set)
+    
```

Comment:
There are still lots of linter warnings throughout this file, lots of long line lengths, plus plenty of extra tabs in the empty lines, unordered imports at the top. I'm not saying to go fix all the files in Aragorn, but this one we can make good.

Reply:
I think it makes more sense to be consistent across different tools. According to Black, which we use for all our other tools, the file is properly formatted now.

Reply:
I agree. This repo has custom black config so should use black.